extensions/gcp/processors/PutGCSObject.h

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <memory>
+
+#include "core/Processor.h"
+#include "core/logging/Logger.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "../controllerservices/GCPCredentialsControllerService.h"
+#include "google/cloud/storage/client.h"
+#include "google/cloud/storage/retry_policy.h"
+#include "utils/Enum.h"
+
+namespace org::apache::nifi::minifi::extensions::gcp {
+
+class PutGCSObject : public core::Processor {
+ public:
+  SMART_ENUM(PredefinedAcl,
+             (AUTHENTICATED_READ, "authenticatedRead"),
+             (BUCKET_OWNER_FULL_CONTROL, "bucketOwnerFullControl"),
+             (BUCKET_OWNER_READ_ONLY, "bucketOwnerRead"),
+             (PRIVATE, "private"),
+             (PROJECT_PRIVATE, "projectPrivate"),
+             (PUBLIC_READ_ONLY, "publicRead"),
+             (PUBLIC_READ_WRITE, "publicReadWrite"));
```

Comment:
I was unable to do this because the gcs::PredefinedAcl::AuthenticatedRead.value is not constexpr
`Constexpr variable 'values' must be initialized by a constant expression`